synapse/rest/client/login.py

```diff
@@ -163,15 +163,16 @@ def on_GET(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
     async def on_POST(self, request: SynapseRequest) -> Tuple[int, LoginResponse]:
         login_submission = parse_json_object_from_request(request)
 
-        if self._refresh_tokens_enabled:
-            # Check if this login should also issue a refresh token, as per MSC2918
-            should_issue_refresh_token = login_submission.get(
-                LoginRestServlet.REFRESH_TOKEN_PARAM, False
-            )
-            if not isinstance(should_issue_refresh_token, bool):
-                raise SynapseError(400, "`refresh_token` should be true or false.")
-        else:
-            should_issue_refresh_token = False
+        # Check to see if the client requested a refresh token.
+        client_requested_refresh_token = login_submission.get(
+            LoginRestServlet.REFRESH_TOKEN_PARAM, False
+        )
+        if not isinstance(client_requested_refresh_token, bool):
+            raise SynapseError(400, "`refresh_token` should be true or false.")
+
+        should_issue_refresh_token = (
+            self._refresh_tokens_enabled and client_requested_refresh_token
+        )
```

Comment:
If they request a refresh token but it's not enabled on this server, are clients required to fall back to the old non-refreshable mechanism?

Reply:
Yep! That'd be the same as connecting to a server that doesn't support refresh tokens.